adapters/improvedigital/improvedigital.go

```diff
@@ -276,7 +276,7 @@ func (a *ImprovedigitalAdapter) getAdditionalConsentProvidersUserExt(request ope
 	}
 
 	// Check key exist user.ext.ConsentedProvidersSettings.consented_providers
-	var cpMap = make(map[string]json.RawMessage)
+	var cpMap = make(map[string]interface{})
```

Comment:
@ishihanvcs any particular reason to use `interface{}` here? this change add step for asserting cMapValue to string (line 291)

```
consentStr := cpMapValue.(string)
```

On line 291, adapter code is not checking whether cpMapValue can be asserted into string. This could led to panic at run time. Refer https://go.dev/play/p/BnmZ2ZGjT3E as example


Additionally `json.RawMessage` does gurantee that value parsed in `cpMap` will be a valid json which can be converted further into string.  But with `interface{}` cpMap can have any value